backend/src/GraphQl/OpenIdConnect/Application/ApplicationByIdDataLoader.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using GreenDonut;
+using Metabase.Data;
+using OpenIddict.Core;
+
+namespace Metabase.GraphQl.OpenIdConnect.Application
+{
+    public sealed class ApplicationByIdDataLoader
+    : BatchDataLoader<Guid, OpenIdApplication?>
+    {
+        private OpenIddictApplicationManager<OpenIdApplication> _applicationManager;
+
+        public ApplicationByIdDataLoader(
+            IBatchScheduler batchScheduler,
+            DataLoaderOptions options,
+            OpenIddictApplicationManager<OpenIdApplication> applicationManager) : base(batchScheduler, options)
+        {
+            _applicationManager = applicationManager;
+        }
+
+        protected override async Task<IReadOnlyDictionary<Guid, OpenIdApplication?>> LoadBatchAsync(IReadOnlyList<Guid> keys, CancellationToken cancellationToken)
+        {
+            var ret = new Dictionary<Guid, OpenIdApplication?>();
+            foreach (var key in keys)
+            {
+                var application = await _applicationManager.FindByIdAsync(key.ToString(), cancellationToken: cancellationToken).ConfigureAwait(false);
+                if (application != null)
```

Comment:
Why do you exclude those for which `application` is equal to `null`? The method's return type allows the application to be `null`. And although it isn't mentioned on https://chillicream.com/docs/hotchocolate/v14/fetching-data/dataloader/#batch-dataloader , as far as I remember, the dictionary **must** contain one entry for each `Guid` given in `keys`. So having a key mapped to a `null` value is actually what we want and need. Correct me if I'm wrong and there was a good reason to exclude the `null` mappings.